File: typescript/src/booking/application/integration/payment/adapter/paymentRequest.ts
// Anti-Corruption Layer: traduce gli eventi di dominio booking nei command di integrazione del BC payment.

import type { BookingPlaced } from '@/booking/domain/events/bookingPlaced.js';
import { paymentRequestIntegrationCommand } from '@/payment/integration/paymentRequestIntegrationCommand.js';
import type { PaymentRequestIntegrationCommand } from '@/payment/integration/paymentRequestIntegrationCommand.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export function paymentRequestFromBookingPlaced(
  event: BookingPlaced
): PaymentRequestIntegrationCommand {
  requireArgument(event, 'event');
  return paymentRequestIntegrationCommand(
    event.aggregateId.value.value,
    event.amount
  );
}

File: java/src/main/java/it/giannibombelli/wsc2026/booking/application/integration/payment/adapter/PaymentRequest.java
package it.giannibombelli.wsc2026.booking.application.integration.payment.adapter;

import it.giannibombelli.wsc2026.booking.domain.events.BookingPlaced;
import it.giannibombelli.wsc2026.common.utils.Require;
import it.giannibombelli.wsc2026.payment.integration.PaymentRequestIntegrationCommand;

public final class PaymentRequest {

    private PaymentRequest() {
    }

    public static PaymentRequestIntegrationCommand fromBookingPlaced(BookingPlaced event) {
        Require.requireArgument(event, "event");
        return new PaymentRequestIntegrationCommand(
            event.aggregateId().value().toString(),
            event.amount()
        );
    }
}
